components/brave_sync/network_time_helper.cc

```diff
@@ -31,14 +31,29 @@ void NetworkTimeHelper::SetNetworkTimeTracker(
   ui_task_runner_ = ui_task_runner;
 }
 
+void NetworkTimeHelper::Shutdown() {
+  network_time_tracker_ = nullptr;
+  ui_task_runner_.reset();
+}
+
 void NetworkTimeHelper::GetNetworkTime(GetNetworkTimeCallback cb) {
   if (!network_time_for_test_.is_null()) {
     std::move(cb).Run(network_time_for_test_);
     return;
   }
+
+  if (!ui_task_runner_) {
+    // Safe exit. Either we don't have it set yet, or we are shutting down
+    // the browser.
+    std::move(cb).Run(base::Time::Now());
```

Comment:
What is the effect of returning `Now()` if `network_time_tracker_` isn't set yet? Does this affect sync logic?

Reply:
When the local time is close to the real time - sync works fine even without `network_time_tracker_` call.

If the local time differs significantly - this mode is not supposed. So it should be used as a safe option.

Reply:
actually there is a pretty big problem here if we are using `http://clients2.google.com/time/1/current` cc @ShivanKaul